js/core/utils/deferred.d.ts

```diff
@@ -3,19 +3,19 @@ interface Callback<T> {
 }
 declare class DeferredObj<T> {
     constructor();
-    state(): string;
+    state(): 'pending' | 'rejected' | 'resolved';
     always(alwaysCallback?: Callback<T>): DeferredObj<T>;
     catch(catchCallback?: Callback<T>): DeferredObj<T>;
     then(resolveCallback?: Callback<T>, rejectCallback?: Callback<T>): DeferredObj<T>;
     done(doneCallback?: Callback<T>): DeferredObj<T>;
     fail(failCallback?: Callback<T>): DeferredObj<T>;
     progress(progressCallback?: Callback<T>): DeferredObj<T>;
-    notify(value?: T, ...args: T[]): DeferredObj<T>;
-    notifyWith(context: DeferredObj<T>, args?: T[]): DeferredObj<T>;
-    reject(value?: T, ...args: T[]): DeferredObj<T>;
-    rejectWith(context: DeferredObj<T>, args?: T[]): DeferredObj<T>;
-    resolve(value?: T, ...args: T[]): DeferredObj<T>;
-    resolveWith(context: DeferredObj<T>, args?: T[]): DeferredObj<T>;
+    notify(...args: T[]): DeferredObj<T>;
+    notifyWith(context: DeferredObj<T>, ...args: T[]): DeferredObj<T>;
+    reject(...args: T[]): DeferredObj<T>;
+    rejectWith(context: DeferredObj<T>, ...args: T[]): DeferredObj<T>;
+    resolve(...args: T[]): DeferredObj<T>;
+    resolveWith(context: DeferredObj<T>, ...args: T[]): DeferredObj<T>;
```

Comment:
Changing `?` to `...` breaks methods singatures
```suggestion
    notify(...args: T[]): DeferredObj<T>;
    notifyWith(context: DeferredObj<T>, ?args: T[]): DeferredObj<T>;
    reject(...args: T[]): DeferredObj<T>;
    rejectWith(context: DeferredObj<T>, ?args: T[]): DeferredObj<T>;
    resolve(...args: T[]): DeferredObj<T>;
    resolveWith(context: DeferredObj<T>, ?args: T[]): DeferredObj<T>;
```